src/Classes/GameController.h

```diff
@@ -1,20 +1,29 @@
 #ifndef __GAMECONTROLLER_H__
 #define __GAMECONTROLLER_H__
 
-#include "cocos2d.h"
+#include "MapController.h"
+#include "MoveController.h"
+#include "IGameEventListener.h"
 
-class GameController : public cocos2d::Layer
-{
-public:
-    static cocos2d::Scene* createScene();
+#include "cocos2d.h"
 
-    virtual bool init();
-    
-    // a selector callback
-    void menuCloseCallback(cocos2d::Ref* pSender);
-    
-    // implement the "static create()" method manually
-    CREATE_FUNC(GameController);
-};
+namespace tsg {
+    namespace game{
+        /*
+         Manages game components
+         */
+        class GameController : public cocos2d::Layer {
+        private:
+            map::MapController * mapController;
+            move::MoveController * moveController;
+            std::vector<IGameEventListener *> listeners;
+        public:
+            static cocos2d::Scene* createScene();
+            virtual bool init();
+            CREATE_FUNC(GameController);
+            void registerListener(IGameEventListener * listener);
+        };
+    }
+}
```

Comment:
Could we use some kind of auto-formatting. Like http://clang.llvm.org/docs/ClangFormat.html